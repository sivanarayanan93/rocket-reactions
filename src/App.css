body {
  background-color: #f0f2f5;
}

.App {
  width: 50%;
  height: 100%;
  margin:auto;
  display: grid;
}

.react-tooltip {
  background-color: #161616!important;
  border-radius: 2px !important;
  color: #fff !important;
  font-size: 12px !important;
  padding: 12px 16px !important;
}

.emoji-count {
  cursor: pointer;
  font-size: 15px;
  padding-left: 8px;
}

