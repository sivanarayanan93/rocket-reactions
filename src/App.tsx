import { useEffect } from 'react';
import './App.css';
import Posts from './views/Posts';
import { getUserInfo } from './shared/Users/UsersApi';
import { useDispatch, useSelector } from 'react-redux';
import { TReducers } from './reducers';
import styled from 'styled-components';
import COLORS from './shared/colors';
import { getPosts } from './shared/Posts/PostModel';

const Header = styled.h3`
  background-color: ${COLORS.WHITE};
  margin: 0px;
  margin-bottom: 10px;
  padding: 16px;
`

const posts = getPosts();

function App() {
  const dispatch = useDispatch(),
    currentUser = useSelector<TReducers, TReducers["User"]["currentUser"]>(state => state.User.currentUser);

  useEffect(() => {
    dispatch(getUserInfo());
  }, [dispatch]);
  return (
    <>
      <Header>Rocket Reactions</Header>
      <div className="App">
        {currentUser.id && <Posts contents={posts}/>}
        {!currentUser.id && <h3>Loading...</h3>}
      </div>
    </>
  );
}

export default App;
