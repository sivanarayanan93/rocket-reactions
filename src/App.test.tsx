import { Provider } from 'react-redux';
import store from './store';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const { asFragment } = render(<Provider store={store}><App /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});
