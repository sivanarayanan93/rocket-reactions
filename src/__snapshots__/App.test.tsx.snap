// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <h3
    class="sc-crzoAE ewvRHE"
  >
    Rocket Reactions
  </h3>
  <div
    class="App"
  >
    <h3>
      Loading...
    </h3>
  </div>
</DocumentFragment>
`;
